fs/tarfs.py

```diff
@@ -204,59 +204,55 @@ def __str__(self):
         return "<TarFS '{}'>".format(self._file)
 
     def getinfo(self, path, namespaces=None):
-        self.check()
+        _path = relpath(self.validatepath(path))
         namespaces = namespaces or ()
-        path = relpath(normpath(path))
-        if not path:
-            raw_info = {
-                "basic":
-                {
-                    "name": "",
-                    "is_dir": True,
-                },
-                "details":
-                {
+        raw_info = {}
+
+        if _path in '/':
+            raw_info['basic'] = {
+                "name": "",
+                "is_dir": True,
+            }
+            if 'details' in namespaces:
+                raw_info['details'] = {
                     "type": int(ResourceType.directory)
                 }
-            }
+
         else:
             try:
-                member = self._tar.getmember(path)
+                member = self._tar.getmember(_path)
             except KeyError:
                 raise errors.ResourceNotFound(path)
 
+            raw_info["basic"] = {
+                "name": basename(member.name),
+                "is_dir": member.isdir(),
+            }
 
-            raw_tar_info = member.get_info(*(
-                [self.encoding, None] if six.PY2 else []
-            ))
-
-            raw_tar_info.update({
-                k.replace('is', 'is_'):getattr(member, k)()
-                for k in dir(member)
-                if k.startswith('is')
-            })
-            raw_info = {
-                "basic":
-                {
-                    "name": basename(member.name),
-                    "is_dir": member.isdir(),
-                },
-                "details":
-                {
+            if "details" in namespaces:
+                raw_info["details"] = {
                     "size": member.size,
                     "type": int(self.type_map[member.type]),
                     "modified": member.mtime,
-                },
-                "access":
-                {
+                }
+            if "access" in namespaces:
+                raw_info["access"] = {
                     "gid": member.gid,
                     "group": member.gname,
                     "permissions": Permissions(mode=member.mode).dump(),
                     "uid": member.uid,
                     "user": member.uname,
-                },
-                "tar": raw_tar_info,
-            }
+                }
+            if "tar" in namespaces:
+                raw_info["tar"] = raw_tar_info = member.get_info(*(
```

Comment:
Purely subjective, but I had to read this line twice. How about this as an alternative?

```python
raw_info["tar"] = raw_tar_info = (
    member.get_info(self.encoding, None)
    if six.PY2 else
    member.get_info()
)
```

Reply:
Indeed, I have no idea why I did that in the first place when I PR'ed TarFS 😄 
IMO the best thing to do would be to define an external `getinfo(member, encoding)` function depending on the Python version once, to avoid checking the Python version on every getinfo call.